ddtrace/_monkey.py

```diff
@@ -167,23 +169,25 @@ def on_import(hook):
                 raise
             error_msg = "failed to import ddtrace module %r when patching on import" % (path,)
             log.error(error_msg, exc_info=True)
-            telemetry.telemetry_writer.add_integration(module, False, PATCH_MODULES.get(module) is True, error_msg)
-            telemetry.telemetry_writer.add_count_metric(
-                "tracers", "integration_errors", 1, (("integration_name", module), ("error_type", type(e).__name__))
-            )
+            if config._telemetry_enabled:
+                telemetry.telemetry_writer.add_integration(module, False, PATCH_MODULES.get(module) is True, error_msg)
+                telemetry.telemetry_writer.add_count_metric(
+                    "tracers", "integration_errors", 1, (("integration_name", module), ("error_type", type(e).__name__))
+                )
         else:
             imported_module.patch()
-            if hasattr(imported_module, "get_versions"):
-                versions = imported_module.get_versions()
-                for name, v in versions.items():
+            if config._telemetry_enabled:
+                if hasattr(imported_module, "get_versions"):
+                    versions = imported_module.get_versions()
+                    for name, v in versions.items():
+                        telemetry.telemetry_writer.add_integration(
+                            name, True, PATCH_MODULES.get(module) is True, "", version=v
+                        )
+                else:
+                    version = imported_module.get_version()
                     telemetry.telemetry_writer.add_integration(
-                        name, True, PATCH_MODULES.get(module) is True, "", version=v
+                        module, True, PATCH_MODULES.get(module) is True, "", version=version
```

Comment:
this wasn't changed, I think github is mis-highlighting here, we simply wrapped the whole block in a conditional